bin/python/ripple/util/test_Sign.py

```diff
@@ -0,0 +1,93 @@
+        self.assertEquals(SField.field_code(SField.STI_VL, 6), 'v')
+
+    def test_to_bytes(self):
+        self.assertEquals(
+            Sign.to_bytes_bigend(12345, 16),
+            '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0009')
+
+    def test_strvl(self):
+        self.assertEquals(Sign.prepend_length_byte(BINARY),
+                          '\x10nN9kfUnKTf7PpgLG')
+
+    def urandom(self, bytes):
+        return '\5' * bytes
+
+    def test_make_seed(self):
+        self.assertEquals(Sign.make_seed(self.urandom),
+                          '\5\5\5\5\5\5\5\5\5\5\5\5\5\5\5\5')
+
+    def test_make_ed(self):
+        private, public = Sign.make_ed25519_keypair(self.urandom)
+        self.assertEquals(private,
+                          '\5\5\5\5\5\5\5\5\5\5\5\5\5\5\5\5'
+                          '\5\5\5\5\5\5\5\5\5\5\5\5\5\5\5\5')
+        self.assertEquals(public,
+                          'nz\x1c\xdd)\xb0\xb7\x8f\xd1:\xf4\xc5Y\x8f\xef\xf4'
+                          '\xef*\x97\x16n<\xa6\xf2\xe4\xfb\xfc\xcd\x80P[\xf1')
+
+    def test_make_manifest(self):
+        _, pk = Sign.make_ed25519_keypair(self.urandom)
+        m = Sign.make_manifest(pk, 'verify', 12345)
+        self.assertEquals(
+            m, '$\x00\x0009q nz\x1c\xdd)\xb0\xb7\x8f\xd1:\xf4\xc5Y\x8f\xef\xf4'
+            '\xef*\x97\x16n<\xa6\xf2\xe4\xfb\xfc\xcd\x80P[\xf1s\x06verify')
+
+    def test_sign_manifest(self):
+        sk, pk = Sign.make_ed25519_keypair(self.urandom)
+        s = Sign.sign_manifest('manifest', sk, pk)
+        self.assertEquals(
+            s, 'manifestv@\xe5\x84\xbe\xc4\x80N\xa0v"\xb2\x80A\x88\x06\xc0'
+            '\xd2\xbae\x92\x89\xa8\'!\xdd\x00\x88\x06s\xe0\xf74\xe3Yg\xad{$'
+            '\x17\xd3\x99\xaa\x16\xb0\xeaZ\xd7]\r\xb3\xdc\x1b\x8f\xc1Z\xdfHU'
+            '\xb5\x92\xac\x82jI\x02')
+
+    def test_wrap(self):
+        wrap = lambda s: Sign.wrap(s, 5)
+        self.assertEquals(wrap(''), '')
+        self.assertEquals(wrap('12345'), '12345')
+        self.assertEquals(wrap('123456'), '123\n456')
+        self.assertEquals(wrap('12345678'), '1234\n5678')
+        self.assertEquals(wrap('1234567890'), '12345\n67890')
+        self.assertEquals(wrap('12345678901'), '123\n456\n789\n01')
+
+    def test_create_ed_keys(self):
+        pkh, skh = Sign.create_ed_keys(self.urandom)
+        self.assertEquals(
+            pkh, 'nHUUaKHpxyRP4TZZ79tTpXuTpoM8pRNs5crZpGVA5jdrjib5easY')
+        self.assertEquals(
+            skh, 'pnEp13Zu7xTeKQVQ2RZVaUraE9GXKqFtnXQVUFKXbTE6wsP4wne')
+
+    def test_get_signature(self):
+        validating = (Base58.VER_NODE_PUBLIC, '\x02' + (32 * 'v'))
+        private = (Base58.VER_NODE_PRIVATE, 32 * 'k')
+
+        Sign.check_validator_public(*validating)
+        Sign.check_master_secret(*private)
+
+        public = Base58.encode_version(*validating)
+        private = Base58.encode_version(*private)
+
+        # TODO: why can't I just do
```

Comment:
Aha.  No, I hadn't really understood that.  I'm going to comment the code more with this information...